pkg/utils/k8s/k8s.go

```diff
@@ -61,6 +61,16 @@ func (l StringMap) ToMap() map[string]string {
 	return l
 }
 
+func (l StringMap) AddIfNotEmpty(k, v string) StringMap {
+	if k == "" {
+		return StringMap{}
+	}
+	if v != "" {
+		l[k] = v
+	}
+	return l
```

Comment:
nit: please feel free to ignore
```suggestion
	if k != "" && v != "" {
		l[k] = v
	}
	return l
```